### PR TITLE
Add psalm as a dev dependency

### PR DESCRIPTION
**Pull request by NicolasCARPi**
Rebased as per https://github.com/nunomaduro/collision/pull/84#issuecomment-579018368

### CODE REVIEW

**Comment by nunomaduro** [comment]
@NicolasCARPi Thanks, can you also add it to the composer test script?

**Comment by nunomaduro** [comment]
I will do it.

**Comment by NicolasCARPi** [comment]
> can you also add it to the composer test script?

Added. `test:static` sounds like a good name to me :)

File: composer.json [hunk 1]
```diff
@@ -60,10 +60,12 @@
     },
     "scripts": {
         "test:types": "phpstan analyse --ansi",
+        "test:static": "psalm",
```
Comment (nunomaduro) [inline composer.json]: Maybe test:types should call `phpstan analyse --ansi && psalm`.

**Review by nunomaduro — CHANGES_REQUESTED** [review]
Maybe test:types should call `phpstan analyse --ansi && psalm`.

**Comment by NicolasCARPi** [comment]
Modified!

**Comment by nunomaduro** [comment]
@NicolasCARPi can you fix tests.

**Comment by NicolasCARPi** [comment]
@nunomaduro Not really no. The errors reported by psalm require careful consideration. I am not familiar enough with the codebase to address all these changes. This PR is only about adding psalm to the project so it can be run easily by a dev, as a first step before fixing the issues reported.

I would suggest disabling it from CI until all issues have been resolved. Once it is not throwing any error, enable it again in CI so PRs will get checked automagically.

**Comment by nunomaduro** [comment]
Let's delay this.

**Pull request closed without merging**